rust/core-lib/src/view.rs

```diff
@@ -874,22 +876,37 @@ impl View {
         self.set_selection_for_edit(text, new_sel);
     }
 
-    pub fn do_find(&mut self, text: &Rope, chars: Option<String>, case_sensitive: bool,
-                   is_regex: bool) {
-        let mut from_sel = false;
-        let search_string = if chars.is_some() {
-            chars
-        } else {
-            self.sel_regions().last().and_then(|region| {
-                if region.is_caret() {
-                    None
+    fn selection_for_find(&mut self, text: &Rope, case_sensitive: bool) {
+        // set last selection or word under current cursor as search query
+        let search_query = match self.selection.last() {
+            Some(region) => {
+                if !region.is_caret() {
+                    text.slice_to_string(region.min(), region.max())
                 } else {
-                    from_sel = true;
-                    Some(text.slice_to_string(region.min(), region.max()))
+                    let (start, end) = {
+                        let mut word_cursor = WordCursor::new(text, region.max());
+                        word_cursor.select_word()
+                    };
+                    text.slice_to_string(start, end)
                 }
-            })
+            },
+            _ => return
         };
 
+        self.find_changed = FindStatusChange::All;
+        self.set_dirty(text);
+
+        // todo: this will be changed once multiple queries are supported
+        // todo: for now only a single search query is supported however in the future
+        // todo: the correct Find instance needs to be updated with the new parameters
+        if self.find.is_empty() {
+            self.find.push(Find::new());
+        }
+
+        self.find.first_mut().unwrap().do_find(text, search_query, case_sensitive, false);
```

Comment:
I'm curious about whether the `selection_for_find` action should always initiate a find. It seems to me that in our investigations a few weeks ago, the editor we were looking at as models didn't always. But I don't think this is a critical question, it seems we can come back to this after the functionality is in place.

Reply:
I added it because, `selection_for_find` could be followed by a `find_next`. But there might be a better way to also handle special cases. I can see if I can come up with something in a future PR.